### PR TITLE
Fixes #16999: Add a reference documentation about methods writing

### PR DESCRIPTION
**Pull request by Fdall**
https://issues.rudder.io/issues/16999

### CODE REVIEW

File: src/reference/modules/reference/pages/generic_methods_guidelines.adoc [hunk 1]
```diff
@@ -1,6 +1,67 @@
...
+Rudder generic methods are building blocks used in technique editor. For general information, see documentation: https://docs.rudder.io/reference/5.0/usage/technique_editor.html
+
+Generic methods are defined in a stand-alone project with code name https://github.com/Normation/ncf[ncf] available in Normation Github.
+Generic methods are implemented for each Rudder agent, with CFEngine language and engine on Unix-based OS, and with Powershell and DSC for Windows based one.
+
+The project is organized in different parts:
+
+----
+ncf
+├── site     # ncf.io site, not relevant
```
Comment (amousset) [inline src/reference/modules/reference/pages/generic_methods_guidelines.adoc]: Does not exist anymore

File: src/reference/modules/reference/pages/generic_methods_guidelines.adoc [hunk 2]
```diff
@@ -14,3 +75,149 @@
...
+
+.Complete list of Tags
+[options="header"]
+|==================================================================================================================================================================
+|Tag name             | Arguments                     | Mandatory   |Multiple  |Comment
+|name                 | <String>                      | Mandatory   |          | Human-readable name of the method
+|description          | <String>                      | Mandatory   |          | One-line description of the method
+|parameter            | <parameter name> <String>     | Mandatory   |          | One-line description of the parameter
+|parameter_constraint | <parameter name> <Constraint> |             | Multiple | Constraint over the given parameter
+|documentation        | <Mardown String>              | Recommended |          | user documentation
```
Comment (amousset) [inline src/reference/modules/reference/pages/generic_methods_guidelines.adoc]: Markdown

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by Fdall** [comment]
OK, squash merging this PR

**Pull request merged**